model/jpa/src/main/java/org/keycloak/models/jpa/JpaRealmProvider.java

```diff
@@ -302,6 +302,24 @@ public RoleModel getClientRole(ClientModel client, String name) {
         return getRolesStream(query, realm, first, max);
     }
 
+    @Override
+    public Stream<RoleModel> getRolesStream(RealmModel realm, Stream<String> ids, String search, Integer first, Integer max) {
+        TypedQuery<String> query;
+
+        if (search == null) {
+            query = em.createNamedQuery("getRoleIdsFromIdList", String.class);
+        } else {
+            query = em.createNamedQuery("getRoleIdsByNameContainingFromIdList", String.class)
+                    .setParameter("search", search);
+        }
+
+        query.setParameter("realm", realm.getId())
+                .setParameter("ids", ids.collect(Collectors.toList()));
```

Comment:
What way of fixing this would you prefer? I can think of 3 ways:

1. Addition of `Cannot be {@code null}` to each parameter that cannot be `null`, which is true basically for each parameter except for `pagination` and `search`.
2. Addition of `@throws NullPointerException When any of required parameters is {@code null}. Parameters are required unless specified differently in the corresponding comment.` to each method. Maybe we can even name all parameters that cannot be `null` instead of the general comment.
3. Addition of some class level comment (similar to one in point 2^). 

I would probably vote for addition of number 2 to `getRolesStream` method in this PR and then include all other methods in some separate PR from cleanup epic. WDYT?

Reply:
I'd prefer with 1 (or 1+2) as it clearly specifies for each parameter that these are required. Throwing NPE for parameters which _cannot be {@code null}_ could be expected, but perhaps a better behaviour would be to just return empty stream if any of the parameters is {@null}?

Reply:
I don't like the idea of returning an empty stream if some required parameter is `null`. I would rather follow the fail-fast principle. If someone accidentally sets the realm to `null`, while they think they used a realm `realmA` they could think this realm contains no role/clients/groups etc. This might be worked around using some WARN log messages in providers. But still, the other not-so-nice thing is that we would need to add a lot of `if` statements into providers, which could make the code messy. WDYT?

I agree with 1+(2-general comment version). But the question is, how we will approach this change? Should we create a new Jira or do we want to have it in this PR?

Reply:
This seems a misunderstanding. I agree that `realm==null` should induce NPE. It needs to be documented.
As for `ids == null`, I believe that returning an empty stream is consistent to the case when `ids.isEmpty()`, isn't it?
Change in this PR would be good